frontend/src/components/Admin.js

```diff
@@ -1,158 +1,255 @@
-       
-       marginLeft: "10vw",
-       marginRight: "auto",
-       
-       }}>
-
-            <Box className="container-info" sx={{
-              display: "flex",
-              justifyContent: "space-between",
-              
-              flexDirection: "row",
-              alignItems: "center",
-              
-              width: '71vw',
-              height: "13rem",
-              marginLeft: "4vw",
-              marginTop: '1vw',
-              marginRight: "auto",
-             
-
-            }}>
-                  <Box className="account-info" sx={{
-                    fontSize: "1.5rem",
-                    fontWeight: "700",
-                    margin: "5px",
-                  }}> Account Details
-
-                      <Profile/>
-                      
-                    </Box>
-                      
-                    <Box className="address-info" sx={{
-                    fontSize: "1.5rem",
-                    fontWeight: "700",
-                    margin: "5px",
-                  }}> Address
-
-                      <Address/>
-                      
-                      </Box>
-
-            
+  const [userDetails, setUserDetails] = useState("");
+  const [hasError, setHasError] = useState(false);
+  const fetchUserDetails = () => {
+    const token = localStorage.getItem("token");
+    fetch("http://localhost:9000/v1/users/user", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error("Error fetching user details");
+        }
+      })
+      .then((userDetails) => {
+        console.log(userDetails);
+        setUserDetails(userDetails);
+        setHasError(false);
+      })
+      .catch((error) => {
+        console.error(error.message);
+        setHasError(true);
+      });
+  };
+
+  const updateUserDetailsFromServer = async () => {
+    const updatedUserDetails = await fetchUserDetails();
+    setUserDetails(updatedUserDetails);
+  };
+
+  useEffect(() => {
+    fetchUserDetails();
+  }, []);
+
+  const logoutAndRedirect = () => {
+    localStorage.removeItem("token");
+    navigate("/");
+  };
+
+  useEffect(() => {
+    if (hasError) {
+      setTimeout(() => {
+        logoutAndRedirect();
+      }, 3000);
+    }
+  }, [hasError, navigate]);
 
+  return (
+    <>
+      <Typography sx={{ fontSize: "1.5rem", fontWeight: "700", margin: "5vw" }}>
+        <IconButton
+          className="navMenu"
+          size="large"
+          edge="start"
+          sx={{ color: "text.primary" }}
+          onClick={() => navigate("/")}
+        >
+          <ArrowBackIcon sx={{ fontSize: 40 }}></ArrowBackIcon>
+        </IconButton>{" "}
+        Back to Shop
+      </Typography>
+
+      <Box
+        className="container-admin"
+        sx={{
```

Comment:
preferable way of styling would be to put these styles into separate object and then just use it where just like you would use css classes. But this still works

Reply:
@namanrai did the styling for this part.